### PR TITLE
server: v1-compatible Vehicle class

### PR DESCRIPTION
**Pull request by Yiin**
do not merge, not tested yet

### CODE REVIEW

File: server/src/classes/Vehicle.cpp [hunk 1]
```diff
@@ -141,14 +142,17 @@ extern js::Class vehicleClass("Vehicle", &sharedVehicleClass, nullptr, [](js::Cl
...
     tpl.Method<&alt::IVehicle::SetDoorState>("setDoorState");
     tpl.Method<&alt::IVehicle::SetWindowOpened>("setWindowOpened");
     tpl.Method<&alt::IVehicle::SetWheelBurst>("setWheelBurst");
     tpl.Method<&alt::IVehicle::SetWheelDetached>("setWheelDetached");
     tpl.Method<&alt::IVehicle::SetWheelOnFire>("setWheelOnFire");
     tpl.Method<&alt::IVehicle::SetWheelHealth>("setWheelHealth");
     tpl.Method<&alt::IVehicle::SetWheelFixed>("setWheelFixed");
+    tpl.Method<&alt::IVehicle::SetWheelHasTire>("setWheelHasTire");
     tpl.Method<&alt::IVehicle::SetPartDamageLevel>("setPartDamageLevel");
+    tpl.Method<&alt::IVehicle::GetPartBulletHoles>("getPartBulletHoles");
```
Comment (xLuxy) [inline server/src/classes/Vehicle.cpp:155]: This is already present in `SharedVehicle.cpp` on Line ~102

File: shared/src/classes/SharedVehicle.cpp [hunk 2]
```diff
@@ -90,14 +91,15 @@ extern js::Class sharedVehicleClass("SharedVehicle", &entityClass, nullptr, [](j
...
+    tpl.Property<&alt::IVehicle::GetRearWheelVariation>("rearWheelVariation");
 
     tpl.Method<&alt::IVehicle::GetMod>("getMod");
     tpl.Method<&alt::IVehicle::GetModsCount>("getModsCount");
     tpl.Method<&alt::IVehicle::IsExtraOn>("isExtraOn");
     tpl.Method<&alt::IVehicle::GetDoorState>("getDoorState");
     tpl.Method<&alt::IVehicle::IsWindowOpened>("isWindowOpened");
     tpl.Method<&alt::IVehicle::IsWheelBurst>("isWheelBurst");
-    tpl.Method<&alt::IVehicle::DoesWheelHasTire>("getWheelHasTire");
+    tpl.Method<&alt::IVehicle::DoesWheelHasTire>("doesWheelHasTire");
```
Comment (xLuxy) [inline shared/src/classes/SharedVehicle.cpp:102]: This also exists in `SharedVehicle.cpp` on line ~100

**Review by xLuxy — CHANGES_REQUESTED** [review]

File: shared/src/classes/SharedVehicle.cpp (hunk 2 above)
Comment (Yiin) [reply shared/src/classes/SharedVehicle.cpp:102]: this is shared file lol

File: client/js/compatibility/classes/vehicle.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    get handbrakeActive() {
+        return super.isHandbrakeActive;
+    }
+
+    get sirenActive() {
+        return super.isSirenActive;
+    }
+
+    get manualEngineControl() {
+        return super.isManulEngineControl;
```
Comment (nrzull) [inline client/js/compatibility/classes/vehicle.js:25]: super.isManu**a**lEngineControl

**Comment by notmatical** [comment]
so test it 😮‍💨

**Comment by Yiin** [comment]
> so test it 😮‍💨

it's open source, no one is stopping you.

**Comment by LeonMrBonnie** [comment]
Most of the API changes here are wrong, the prefix `is` for properties is only allowed to be used when it is read-only.

**Pull request closed without merging**